tests/unit/test_upload.py

```diff
@@ -116,12 +167,28 @@ def test_runtime_error_raised_if_one_or_more_buckets_not_valid(
             " invalid_bucket_2"
         )
 
-        with pytest.raises(RuntimeError, match=re.escape(expected_error)):
+        with self.subTest("RuntimeError raised"), pytest.raises(
+            RuntimeError, match=re.escape(expected_error)
+        ):
             upload.check_buckets_exist(
-                ["invalid_bucket_1", "invalid_bucket_2"]
+                buckets=["invalid_bucket_1", "invalid_bucket_2"],
+                slack_alert_webhook="my_webhook",
             )
 
-    def test_client_error_raised_when_bucket_does_not_exist(self, mock_client):
+        with self.subTest("Correct Slack message sent"):
+            expected_slack_message = (
+                ":warning:  *S3 Upload*: Error in accessing specified S3"
+                f" buckets!\n\n\t\t{expected_error}"
+            )
+
+            self.assertEqual(
+                mock_slack.call_args[1]["message"],
+                expected_slack_message,
+            )
+
+    def test_client_error_raised_when_bucket_does_not_exist(
+        self, mock_client, mock_slack
+    ):
         mock_client.side_effect = s3_exceptions.ClientError(
             {"Error": {"Code": 1, "Message": "foo"}}, "bar"
```

Comment:
would the expected error below also need slack checking since the code should outputs the slack message? Or is this covered by the test above?